src/DotLiquid/Context.cs

```diff
@@ -709,6 +698,8 @@ private static bool IsKeyValuePair(object obj)
                     Type baseType = valueType.GetGenericTypeDefinition();
                     if (baseType == typeof(KeyValuePair<,>))
                     {
+                        key = valueType.GetRuntimeProperty("Key").GetValue(obj);
+                        value = valueType.GetRuntimeProperty("Value").GetValue(obj);

```

Comment:
As mentioned, I don't see any gain by assigning both by reflection when we need 1 or even none for that execution block. Will get back with actual perf comparisons

Reply:
@microalps this is the last issue with this PR, right?
@robin-parker would it be possible to have more specialized helpers to avoid any performance impact from running reflection which may not be required by the caller?

Reply:
I've reverted to `IsKeyValuePair` and added a new helper method `ObjectExtensionMethods.GetPropertyValue` to extract the Key/Values by reflection when needed.

Reply:
Thanks, I can also confirm this PR has no performance change. Signing off.